### ISSUE TITLE
Add support for Custom Metadata

### ISSUE THREAD

**Issue by abd3** [joeferraro/MavensMate#610]
Custom Metadata is a newish type of metadata that allows us to add individual records of a custom metadata type to our project and to deploy them like other metadata types. Partial support for custom metadata exists in Mavensmate, but this is a request to enhance the capabilities as follows:
- We can currently download existing custom metadata records (in the src/customMetadata folder), but we can't edit and save changes to it. Meaning that the metadata files won't compile to Salesforce. This also implies that we can't add new metadata records via Mavensmate.
- .md files in the src/customMetadata folder should be recognized as a "Salesforce file type" by default in the Mavensmate global settings.

**Comment by ralphcallaway** [comment]
@abd3 first part is in the works which allows editing and saving custom metadata, not sure when the next release will go out, but if you're antsy you can download the source and get a dev version going with this in it

do you think you would really be creating custom metadata records in sublime text? there is a pretty small subset of stuff you can create with mavensmate that's mostly code centric, e.g. you can't create custom objects or custom settings

**Comment by abd3** [comment]
Yes, we definitely want/need to edit and create custom metadata records in
MavensMate. We don't need to create new custom metadata *types*, but each
record of an existing type is stored as a file and we (and others) will
want/need the ability to create and deploy them from MavensMate. Lots of
data load tools including those from sAlesforce don't yet handle custom
metadata well, so MavensMate is our target system for doing bulk data loads
of these records. We'll then be adding them to our Git repo with the code
and deploying to other orgs.

On 6:57PM, Fri, Dec 4, 2015 Ralph Callaway <notifications@github.com> wrote:

> @abd3 <https://github.com/abd3> first part is in the works which allows
> editing and saving custom metadata, not sure when the next release will go
> out, but if you're antsy you can download the source and get a dev version
> going with this in it
>
> do you think you would really be creating custom metadata records in
> sublime text? there is a pretty small subset of stuff you can create with
> mavensmate that's mostly code centric, e.g. you can't create custom objects
> or custom settings
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joeferraro/MavensMate/issues/610#issuecomment-162113107>
> .
>
-- 
Thanks and best wishes,
Andrew

**Comment by ralphcallaway** [comment]
yep, editing is already taken care of in that commit i referenced

will need to test deploying, although i expect if we can save the records that will likely work, will also need to test saving a new custom metadata records

regarding creating, would you just create a new file in the customMetadata directory and then compile those files?

in terms of doing bulk data loads, wouldn't you want to do that with ant?

**Comment by abd3** [comment]
Actually, deploying already works. That's fantastic that editing/compiling
will soon work too. Yes, in terms of creation we just create files in that
directory (using a perl script from a CSV). "Bulk" data load would just
imply compiling or deploying a lot of records. So it's only the
editing/compiling part that we were missing.

On 8:11PM, Fri, Dec 4, 2015 Ralph Callaway <notifications@github.com> wrote:

> yep, editing is already taken care of in that commit i referenced
>
> will need to test deploying, although i expect if we can save the records
> that will likely work, will also need to test saving a new custom metadata
> records
>
> regarding creating, would you just create a new file in the customMetadata
> directory and then compile those files?
>
> in terms of doing bulk data loads, wouldn't you want to do that with ant?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joeferraro/MavensMate/issues/610#issuecomment-162121590>
> .
>
-- 
Thanks and best wishes,
Andrew

**Comment by ralphcallaway** [comment]
excellent, good to know deploy is already good, editing/saving was pretty much there, the extra `.` in the metadata record file names was causing some issues

i'll try moving some records to the folder and compiling to see if there are any issue there, otherwise looks like we're pretty much there

thanks for the suggestion!

**Comment by abd3** [comment]
Hey, thank you all for doing the hard work ! ! We absolutely love MavensMate

On 8:20PM, Fri, Dec 4, 2015 Ralph Callaway <notifications@github.com> wrote:

> excellent, good to know deploy is already good, editing/saving was pretty
> much there, the extra . in the metadata record file names was causing
> some issues
>
> i'll try moving some records to the folder and compiling to see if there
> are any issue there, otherwise looks like we're pretty much there
>
> thanks for the suggestion!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joeferraro/MavensMate/issues/610#issuecomment-162122294>
> .
>
-- 
Thanks and best wishes,
Andrew

**Comment by ralphcallaway** [comment]
so just tested manually creating some files and pushing them to salesforce by saving, compiling the project, both worked, looks like we're good to go

**Comment by abd3** [comment]
Amazing, well done and so prompt

On 8:35PM, Fri, Dec 4, 2015 Ralph Callaway <notifications@github.com> wrote:

> so just tested manually creating some files and pushing them to salesforce
> by saving, compiling the project, both worked, looks like we're good to go
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joeferraro/MavensMate/issues/610#issuecomment-162124633>
> .
>
-- 
Thanks and best wishes,
Andrew

**Comment by abd3** [comment]
Hi @ralphcallaway, it looked like a release that includes this might be delayed a bit, so I tried to download and run the code from GitHub. I didn't spend too much time on it, but I wasn't able to figure out how to run mavensmate from the command line, as so far I've just used the App and Sublime Plugin. If you have any pointers on this I'd be grateful. I'm also available at abd3721 at WellKnownEmailServiceFromGoogle dot com

**Comment by ralphcallaway** [comment]
@abd3 yeah it's not obvious, but luckily not terribly hard (run the server, change a config setting in sublime text), you'll need to on v7+ of the st3 plugin (the one which requires the standalone version of the mavensmate app)

see https://github.com/joeferraro/MavensMate/issues/518

hoping to at some point update the read me to include some details for developers ...
